utils/generator/README.md

```diff
@@ -0,0 +1,2638 @@
+
+Example:
+```shell script
+# <struct name="VehicleDataResult" />
+com/smartdevicelink/proxy/rpc/VehicleDataResult.java
+
+# <function name="AddCommand" messagetype="request" />
+com/smartdevicelink/proxy/rpc/AddCommand.java
+# <function name="AddCommand" messagetype="response" />
+com/smartdevicelink/proxy/rpc/AddCommandResponse.java
+# <function name="OnLanguageChange" messagetype="notification" />
+com/smartdevicelink/proxy/rpc/OnLanguageChange.java
+```
+
+The package definition for Struct or Function classes is:
+```java
+package com.smartdevicelink.proxy.rpc;
+``` 
+
+## The License Header
+
+All files should start from the comment with the license information.
+
+```java
+/*
+ * Copyright (c) 2017 - [year], SmartDeviceLink Consortium, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the SmartDeviceLink Consortium Inc. nor the names of
+ * its contributors may be used to endorse or promote products derived
+ * from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+```
+Where `[year]` in the copyright line is the current year.
+
+## `<enum>`
+
+The name of the class is the value from the `"name"` attribute of `<enum>`.
+
+The class should have the next JavaDoc comment:
+```java
+/**
+ * [description]
+ *
+ * @deprecated
+ * @since SmartDeviceLink [since_version]
+ * @see [see_reference]
+ */
+```
+Where:
+* `[description]` is `<description>` of the current `<enum>`, if exists.
+* `@deprecated` indicates the deprecation state if the `"deprecated"` attribute exists and is "true".
+* `@since` should be present, if the `"since"` attribute exists, and `[since_version]` is the `Major.Minor.Patch` formatted value of this attribute.
```

Comment:
@kshala-ford `<element name="FR-CA" internal_name="FR_CA">` will not have the `@since` JavaDoc as described in the document. Should it appear based on the parent `since` attribute?

Reply:
Not really because @since is added to the class’s inline documentation but it would better appear in the ide documentary. I’m okay to leave as is.